src/main/java/org/klodnicki/service/BookService.java

```diff
@@ -85,4 +91,32 @@ public List<String> prepareListOfBooks(String title, String author) {
     private List<BookInfo> getBooksInfoFromDatabaseByTitleAndAuthor(String title, String author) {
         return findBooksByTitleAndAuthor(title, author);
     }
+
+    private List<BookInfo> getAllBooksFromDatabaseSortByParameter(String parameter) {
+        return bookRepository.findAllBooksSortByParameter(parameter);
+    }
+
+
+    public List<String> prepareListOfAllBooksSortByParameter(String parameter) throws NotFoundInDatabaseException,
+            IllegalArgumentException, SortParameterNotFoundException {
+
+        if (!(parameter.equalsIgnoreCase("title") || parameter.equalsIgnoreCase("author")
```

Comment:
This can be greatly improved!
You're not taking into account that there could be in the future another fields on which you will sort. This makes the code not clean, smell bad and it will be harder to maintain.

Reply:
you are right! I made great improvement as we talked. I hope you will like it as much as me ;)